unittests/api_tests.cpp

```diff
@@ -935,6 +946,54 @@ BOOST_AUTO_TEST_CASE(checktime_fail_tests) { try {
    BOOST_REQUIRE_EQUAL( t.validate(), true );
 } FC_LOG_AND_RETHROW() }
 
+BOOST_AUTO_TEST_CASE(checktime_pause_test) { try {
+
+   fc::temp_directory tempdir;
+   auto conf_genesis = tester::default_config( tempdir );
+
+   auto& cfg = conf_genesis.second.initial_configuration;
+
+   cfg.max_block_cpu_usage        = 250'000;
+   cfg.max_transaction_cpu_usage  = 4'999; // needs to be large enough for create_account and set_code
+   cfg.min_transaction_cpu_usage  = 1;
+
+   tester t( conf_genesis.first, conf_genesis.second );
+   t.execute_setup_policy( setup_policy::full );
+
+   t.produce_blocks(2);
+
+   ilog( "create account" );
+   t.create_account( "pause"_n );
+   ilog( "set code" );
+   t.set_code( "pause"_n, contracts::test_api_wasm() ); // only test that should use pause account so it is not cached
+   ilog( "produce block" );
+   t.produce_blocks(1);
+
+   int64_t ram_bytes; int64_t net; int64_t cpu;
+   auto& rl = t.control->get_resource_limits_manager();
+   rl.get_account_limits( "pause"_n, ram_bytes, net, cpu );
+   BOOST_CHECK_EQUAL( cpu, -1 );
+   auto cpu_limit = rl.get_block_cpu_limit();
+   idump(("cpu_limit")(cpu_limit));
+   BOOST_CHECK( cpu_limit <= 250'000 );
+
+   // 6ms should be lower than time to load wasm
```

Comment:
Having deadlines of 4ms, 5ms, and 6ms seems to make this test very dependent on the time required to compile wasm.  I'm seeing times under 6500 µs (i.e. failing the dur > 6500 check), so it wouldn't take a very much faster processor to invalidate the test results.

Reply:
Hard to argue with this assessment. I'm open for suggestions. Maybe add additional comments about this may fail on faster hardware. Could just remove the test now that I have proven to myself at least that this is working.